### PR TITLE
feat: add internal fax server with SRFax provider support

### PR DESCRIPTION
**Pull request by yingbull**
## Summary

- **Provider abstraction layer**: `FaxProviderClient` interface with `SRFaxProviderClient` and `MiddlewareFaxProviderClient` implementations, routed via `FaxProviderClientFactory`
- **SRFax direct integration**: Send, receive, status polling, and mark-as-read via SRFax HTTP API with SSRF protection on configurable endpoint URL
- **Three-phase inbound import**: Download → mark-as-read → import strategy in `FaxImporter` preventing both data loss and duplicate imports
- **Scheduler auto-recovery**: `FaxSchedulerJob` distinguishes OOM/JVM errors (no restart) from transient failures (auto-restart with 10-minute delay)
- **Security hardening**: Path traversal protection via `PathValidationUtils`, encrypted credential storage with `EncryptionUtils`, password leak fix in `ConfigureFax2Action`, detached entity copies in `getFaxGatewayAccounts()`
- **Spring DI migration**: All fax core services (`FaxSender`, `FaxImporter`, `FaxStatusUpdater`) use `@Autowired` constructor injection instead of `SpringUtils.getBean()`
- **Transient error retry**: `FaxProviderException.isTransient()` classification allows network errors to revert to WAITING status for automatic retry
- **FaxSender fail-fast**: Throws `IllegalStateException` when DOCUMENT_DIR is not configured instead of silently dropping all outbound faxes

## Test plan

- [x] 94 fax module tests passing (0 failures, 3 @Disabled with actionable descriptions)
- [x] `FaxSenderTest`: 19 tests covering send workflow, resolveFilePath path traversal security, DOCUMENT_DIR validation, transient error retry
- [x] `FaxStatusUpdaterTest`: 10 tests covering status polling, orphaned fax detection, merge failure isolation
- [x] `FaxImporterPollTest`: 6 tests covering poll cycle, retry logic, error resilience
- [x] `FaxImporterCriticalGapsTest`: 13 tests covering PDF validation, collision-free filenames
- [x] `SRFaxProviderClientTest`: 20 tests covering status mapping and FaxDetailsID pipe parsing
- [x] `SRFaxProviderClientEnsureSuccessTest`: 13 tests covering API response parsing edge cases
- [x] `SRFaxProviderClientSsrfTest`: SSRF protection for configurable SRFax URL
- [x] `MiddlewareFaxProviderClientValidationTest`: Input validation for middleware provider
- [x] `FaxProviderClientFactoryTest`: 6 tests covering factory routing and duplicate detection
- [ ] Manual: Configure SRFax account in admin UI, send outbound fax, verify delivery
- [ ] Manual: Verify inbound fax download and import to document queue

🤖 Generated with [Claude Code](https://claude.com/claude-code)

___

### **Description**
- Enhanced the fax configuration admin interface with improved security and usability features.
- Implemented a robust three-phase import strategy in the FaxImporter to prevent data loss.
- Added comprehensive unit tests for the FaxSender to ensure reliability and error handling.
- Introduced a SQL migration pattern enforcer to maintain database integrity.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>configureFax.jsp</strong><dd><code>Enhance Fax Configuration Admin Interface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/main/webapp/admin/configureFax.jsp
<li>Added copyright headers and improved documentation.<br> <li> Enhanced security features including CSRF protection and path <br>validation.<br> <li> Improved user interface with better styling and unsaved changes <br>warning.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/431/files#diff-a8855cf727b299261610b13fd74d9597522dbd5f9785a37880934f6bf2ec83ea">+427/-263</a></td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>FaxImporter.java</strong><dd><code>Improve Fax Importer Logic and Documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/main/java/io/github/carlos_emr/carlos/fax/core/FaxImporter.java
<li>Added detailed documentation for the FaxImporter class.<br> <li> Implemented three-phase import strategy for handling faxes.<br> <li> Improved error handling and logging for better traceability.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/431/files#diff-c378972221eb4672522ea5a2463b5e6cbdfaebea20c1684be02dbcd112f02f4e">+612/-194</a></td>
</tr>

<tr>
  <td>
    <details>
      <summary><strong>enforce-sql-migration-pattern.py</strong><dd><code>SQL Migration Pattern Enforcer Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.claude/hooks/enforce-sql-migration-pattern.py
<li>Implemented a hook to enforce SQL migration standards.<br> <li> Prevents modification of original SQL files and ensures patch file <br>validity.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/431/files#diff-8d439ff2a25c6be7af5f498890125d558b830cc2b1beb808c4933f87a61ff55a">+183/-0</a>&nbsp; </td>
</tr>
</table></td></tr><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>FaxSenderTest.java</strong><dd><code>Add Unit Tests for FaxSender Functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/test-modern/java/io/github/carlos_emr/carlos/fax/core/FaxSenderTest.java
<li>Added unit tests for the FaxSender class.<br> <li> Covered various scenarios including error handling and transient <br>errors.<br> <li> Ensured proper logging and status updates for fax jobs.<br>

</details>

  </td>
  <td><a href="https://github.com/carlos-emr/carlos/pull/431/files#diff-8ee30b76b6653e76c8dd29db7b602aa8de3d7da8ad8d43eb57688d882e4020da">+692/-0</a>&nbsp; </td>
</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **Penify usage**:
>Comment `/help` on the PR to get a list of all available Penify tools and their descriptions

## Summary by CodeRabbit

* **New Features**
  * Added support for multiple fax providers (SRFax and Middleware relay) with provider-type selection in fax configuration.
  * Enhanced fax scheduler with automatic restart capability and improved status reporting.
  * Added incoming fax queue visibility and pending fax listing in admin console.

* **Configuration**
  * New fax provider configuration options and settings.
  * Added document recycle bin and update restriction properties.
  * Refined fax admin permissions for improved security control.

* **Documentation**
  * Added comprehensive fax provider configuration guide.

### CODE REVIEW

File: src/main/webapp/admin/configureFax.jsp [hunk 1]
```diff
@@ -246,204 +364,250 @@
...
-                <div class="row" id="user<%=count == 0 ? "" : count%>">
+                <div class="row fax-card">
                     <div class="span12">
+                        <%
+                            // Current UI supports configuring one fax account at a time.
+                            // Backend array-based save infrastructure is preserved for potential future multi-account support.
+
+                            // Get first config or create defaults for empty config
+                            // Note: "config" is a JSP implicit object (ServletConfig), so use "faxCfg" instead
+                            FaxConfig faxCfg = faxConfigList.isEmpty() ? null : faxConfigList.get(0);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/admin/configureFax.jsp:443]: current intended function, only one configured at any given time currently

File: src/main/java/io/github/carlos_emr/carlos/fax/core/FaxImporter.java [hunk 2]
```diff
@@ -58,277 +51,701 @@
...
+                        } catch (IOException e) {
+                            retryFax.setStamp(new Date());
+                        }
 
-        filename = filename.replace("|", "-");
+                        EDoc edoc = importFromIncoming(pdfFile, faxConfig, retryFax);
+                        if (edoc != null) {
+                            try {
+                                providerRouting(Integer.parseInt(edoc.getDocId()));
+                            } catch (RuntimeException e) {
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/java/io/github/carlos_emr/carlos/fax/core/FaxImporter.java:536]: will look at middleware next update

File: src/main/java/io/github/carlos_emr/carlos/commn/model/FaxConfig.java [hunk 3]
```diff
@@ -34,149 +31,155 @@
...
                 }
                 // Legacy plain text - return as-is, caller decides whether to re-encrypt
                 return value;
+            } catch (Exception e) {
+                logger.error("Failed to decrypt {} - possible key rotation or data corruption. "
+                        + "Re-enter the password in Administration > Faxes > Configure Fax to "
+                        + "re-encrypt with the current key.", fieldLabel, e);
+                throw new IllegalStateException(
+                        "Failed to decrypt " + fieldLabel
+                        + " - re-enter password in Administration > Faxes > Configure Fax", e);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/java/io/github/carlos_emr/carlos/commn/model/FaxConfig.java:181]: ok for now

File: src/main/java/io/github/carlos_emr/carlos/fax/core/FaxImporter.java [hunk 4]
```diff
@@ -58,277 +51,701 @@
...
+            .replace(".TIF", ".pdf")
+            .replace(".TIFF", ".pdf");
+
+        if (!sanitized.toLowerCase().endsWith(".pdf")) {
+            sanitized += ".pdf";
         }
 
-        log.debug("Failed to save file " + filename + " to filesystem at " + DOCUMENT_DIR);
+        return sanitized;
+    }
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/java/io/github/carlos_emr/carlos/fax/core/FaxImporter.java:683]: will leave for now

File: src/main/java/io/github/carlos_emr/carlos/fax/core/FaxSchedulerJob.java [hunk 5]
```diff
@@ -1,131 +1,334 @@
...
+                                attempt, schedulingEx);
+                        lastError.set("Auto-restart scheduling failed at attempt " + attempt + " - manual restart required");
+                    }
+                }
+            }
+        };
+
+        logger.error("Fax scheduler auto-restart scheduled: attempt {} in {} ms", attempt, FAILURE_RESTART_DELAY_MS);
+        timer.schedule(autoRestartTask, FAILURE_RESTART_DELAY_MS);
     }
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/java/io/github/carlos_emr/carlos/fax/core/FaxSchedulerJob.java:229]: will look at next version

**Comment by phc007** [comment]
@claude the service fails and prevents carlos.war from loading if it cannot find /var/lib/tomcat9/fax-incoming
this is problematic in two ways
1. The entire program fails if the fax service fails even if you are not using the fax service
2. the location is hard coded in a non typical place, rather than taken from carlos.properties, with a default relating to the document structure there
Please suggest improvements
